### DISCUSSION TITLE
ISMRMRD Conversion Questions

### DISCUSSION THREAD

**Discussion by alexjaffray** [facebookresearch/fastMRI#189] [Q&A] [answered]
Hello fastMRI community,

I am working with image reconstructions in Julia (specifically using the MRIReco.jl package). I noticed that all of the dataset is provided in ISMRMRD format, and was wondering what utility was used to do the conversion from native raw data formats (.raw,.dat, etc...) to ISMRMRD. Can anyone provide some comments on a workflow that is robust and consistent for data conversion between vendor formats and ISMRMRD?

Thanks,
Alexander

**Comment by mmuckley** [comment]
Hello @alexjaffray, in our case we used the [ISMRMRD implementation on GitHub](https://github.com/ismrmrd/ismrmrd). I don't remember exactly but I think we used the MATLAB interface.

Also pinging @dchansen who might have better advice based on more experience.

**Comment by dchansen** [comment, marked as answer]
Hey @alexjaffray 
The [ISMRMRD account](https://github.com/ismrmrd/) has projects for converting from Siemens, GE, Philips and Bruker MRI systems. 
The Siemens converter is the most fully features, but I know that several vendors (such as GE) have converters available on their research forums. 

If you need to mess around with the ISMRMRD files, the Python interface is probably the most ergonomic.

**Reply by alexjaffray** [reply]
Thank you and @mmuckley for the quick response! I have worked with the Siemens converter in the past and found it very intuitive to use and fully featured. I am now working with Philips data, and was wondering if there is active development to bring all of the converter utilities to feature parity. I guess I will play around with the Philips converter code to see if I can make it work for my applications.

**Reply by dchansen** [reply]
Unfortunately no. The Siemens and GE research communities have been quite active in developing these converters, with some support from the vendors themselves. 
We would be happy to support a more fully featured Philips converter, but someone from the Philips community (or Philips themselves) would have to do the majority of the development.
From speaking with Philips users, this should be a relatively straight forward task if you have access to their research framework (PARADISE), and could even support online streaming and reconstruction of the data.
